samtranslator/model/api/api_generator.py

```diff
@@ -513,16 +513,18 @@ def _construct_api_domain(self, rest_api, route53_record_set_groups):  # type: i
             basepath_mapping.Stage = ref(rest_api.logical_id + ".Stage")
             basepath_resource_list.extend([basepath_mapping])
         else:
-            for path in basepaths:
-                path = "".join(e for e in path if e.isalnum())
+            for basepath in basepaths:
+                # ignore leading and trailing `/` in the path name
```

Comment:
Why?

Reply:
`A base path may only contain letters, numbers, and one of "$-_.+!*'()"`

Reply:
That doesn't explain why it's only stripping slashes.